### PR TITLE
TESB416-283 Refactoring to make Sonar happy

### PR DESCRIPTION
**Pull request by flitzpiepe96**

### CODE REVIEW

File: data api/src/main/java/de/tinf15b4/kino/data/ServiceImplModel.java [hunk 1]
```diff
@@ -53,24 +52,24 @@ public E findOne(long id) {
...
+                String table = updated.getClass().getAnnotation(Table.class).name();
 
                 // detach old entity
                 entityManager.flush();
-                entityManager.detach(e);
+                entityManager.detach(updated);
 
                 // adjust the id
-                entityManager.createNativeQuery("UPDATE " + table + " SET id = " + oldid + " WHERE id = " + newid).executeUpdate();
+                entityManager.createNativeQuery(String.format("UPDATE %s SET id = %s WHERE id = %s", table, oldid, newid)).executeUpdate();
```
Comment (rgcjonas) [inline data api/src/main/java/de/tinf15b4/kino/data/ServiceImplModel.java]: I'm inclined to believe that sonarqube had another resolution of this alleged SQL injection vulnerability in mind.

If it makes sonarqube shut up then I guess we'll keep it, but you really could have gone the extra mile here and make use of proper query parameters.

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java [hunk 2]
```diff
@@ -46,33 +48,33 @@ public AbstractCinemaScraper() {
 
     public void scrape() {
         logger = getLogger();
-        logger.info(String.format("Initializing Webdriver for scraper: [%s]", cinema.getName()));
+        logger.info("Initializing Webdriver for scraper: [%s]", cinema.getName());
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: I didn't even know this was possible :D

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java [hunk 3]
```diff
@@ -68,7 +65,7 @@ protected GatheringResult gatherData() {
                         String datestr = e.getAttribute("data-original-title");
 
                         // BAD: Hardcode prices
-                        Map<DayOfWeek, Integer> priceMap = new HashMap<>();
+                        Map<DayOfWeek, Integer> priceMap = new EnumMap<>(DayOfWeek.class);
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java]: Just curious: Why is `EnumMap` preferred here? Performance?

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java [hunk 4]
```diff
@@ -103,7 +100,7 @@ protected Cinema getCinema() {
         WebElement container = bilderEl.findElement(By.xpath(".."));
         WebElement imgEl = container.findElement(By.cssSelector("img"));
 
-        byte imgBytes[];
+        byte[] imgBytes;
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java]: Old C habits die hard... Sorry

File: commons/src/main/java/de/tinf15b4/kino/data/cinemas/Cinema.java [hunk 5]
```diff
@@ -58,7 +67,7 @@ public Cinema(String name, String street, String hnr, String postcode, String ci
     }
 
     public String getAddress() {
-        return String.format("%s %s\n%s %s, %S", street, hnr, postcode, city, country);
+        return String.format("%s %s%n%s %s, %S", street, hnr, postcode, city, country);
```
Comment (flitzpiepe96) [inline commons/src/main/java/de/tinf15b4/kino/data/cinemas/Cinema.java]: Exactly what you mentioned: The reason is that each platform users the corresponding linebreak

File: commons/src/main/java/de/tinf15b4/kino/data/favorites/Favorite.java [hunk 6]
```diff
@@ -16,9 +16,14 @@
...
 public class Favorite extends EntityModel implements ImageContainer {
 
-    public interface FieldInfos {
-        String USER = "user";
-        String CINEMA = "cinema";
+    static class FieldInfos {
+        private FieldInfos() {
+            // Just used to hide the public constructor
+            // Sonarqube seems to like that
+        }
```
Comment (flitzpiepe96) [inline commons/src/main/java/de/tinf15b4/kino/data/favorites/Favorite.java]: We could easily just mark the errors as won't fix. I just did it becuase it was easy

File: data api/src/main/java/de/tinf15b4/kino/data/ServiceImplModel.java (hunk 1 above)
Comment (flitzpiepe96) [inline data api/src/main/java/de/tinf15b4/kino/data/ServiceImplModel.java]: Sure the intention was to use the predefined query thingy. I just could figure out how to do it in this context ^^
I found no method that allowed to do something like taht. Probably should have checked in more details here

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java (hunk 2 above)
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: As did I. But apparently this only evaluates the string and all parameters if needed. If the logging level does not show the message it doesnt eval it. So actually its pretty nice :D

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java (hunk 3 above)
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java]: I dont know but probably yes. I guess enummap has a fixed size as the enum count is already known. And fixed size drastically speeds up things as we know (List <-> Array)

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java (hunk 4 above)
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/ZkmFilmpalastScraper.java]: Wel java allows doing it the way you did so thats fine with me, but not with sonar it looks like :D

**Pull request merged**